sklearn/calibration.py

```diff
@@ -311,145 +370,202 @@ def _more_tags(self):
         }
 
 
-class _CalibratedClassifier:
-    """Probability calibration with isotonic regression or sigmoid.
+def get_prediction_method(clf_fitted, return_string=False):
+    """Return prediction method or their corresponding name as string.
 
-    It assumes that base_estimator has already been fit, and trains the
-    calibration on the input set of the fit function. Note that this class
-    should not be used as an estimator directly. Use CalibratedClassifierCV
-    with cv="prefit" instead.
+    `decision_function` method of `clf_fitted` returned, if it
+    exists, otherwise `predict_proba` method returned.
 
     Parameters
     ----------
-    base_estimator : instance BaseEstimator
-        The classifier whose output decision function needs to be calibrated
-        to offer more accurate predict_proba outputs. No default value since
-        it has to be an already fitted estimator.
+    clf_fitted : Estimator instance
+        Classifier to obtain the prediction method from.
 
-    method : 'sigmoid' | 'isotonic'
-        The method to use for calibration. Can be 'sigmoid' which
-        corresponds to Platt's method or 'isotonic' which is a
-        non-parametric approach based on isotonic regression.
+    return_string : bool, default=False
+        Whether to return the method name as string instead of the prediction
+        method.
+
+    Returns
+    -------
+    prediction_method : callable or str
+        If `return_string=True`, name of the prediction method as string.
+        If `return_string=False`, the prediction method.
+    """
+    if hasattr(clf_fitted, "decision_function"):
+        method_name = "decision_function"
+        method = getattr(clf_fitted, "decision_function")
+    elif hasattr(clf_fitted, "predict_proba"):
+        method_name = "predict_proba"
+        method = getattr(clf_fitted, "predict_proba")
+    else:
+        raise RuntimeError("'base_estimator' has no 'decision_function' or "
+                           "'predict_proba' method.")
+    if return_string:
+        return method_name
+    else:
+        return method
 
-    classes : array-like, shape (n_classes,), optional
-            Contains unique classes used to fit the base estimator.
-            if None, then classes is extracted from the given target values
-            in fit().
 
-    See also
-    --------
-    CalibratedClassifierCV
+def _reshape_preds(preds, method, n_classes):
+    """Reshape predictions when classification binary.
```

Comment:
missing "is"?

But also this seems to reshape regardless of whether the problem is binary or not?

Reply:
also since this only seems to be used in _get_predictions maybe we can just declare it within `_get_predictions` as a subfunction, or just write the code "inline".